xtask/codegen/src/yaml_kinds_src.rs

```diff
@@ -0,0 +1,35 @@
+use crate::kind_src::KindsSrc;
+
+pub const YAML_KINDS_SRC: KindsSrc = KindsSrc {
+    punct: &[
+        (":", "COLON"),
+        (",", "COMMA"),
+        ("{", "L_CURLY"),
+        ("}", "R_CURLY"),
+        ("[", "L_BRACK"),
+        ("]", "R_BRACK"),
+        ("-", "DASH"),
+        ("%", "PERCENT"),
+        ("*", "ASTERISK"),
+        ("#", "HASH"),
+        ("---", "DOC_START"),
+        ("...", "DOC_END"),
```

Comment:
Yea that sounds good to me, I overlooked them.

Reply:
https://yaml.org/spec/1.2.2/#anchors-and-aliases searching in this file for `override` and `<<` gave me no results, do you know why that is different than the atlassian site?

Reply:
I'm a bit surprised, I used overrides myself when dealing with Gitlab CI configurations (so it's not only Atlassian who uses them). Looking into that now... 🤔

Reply:
Surely yaml.org would keep their own spec updated right ..... right?

Reply:
Oh. Interesting. See https://yaml.org/spec/1.2.2/ext/changes/ , in section "Changes in version 1.2 (revision 1.2.0) (2009-07-21)".

Reply:
Okay that would mean that in our quest of supporting the latest (for now) we should leave it out. Although that immediatly makes it less useful since most people use yaml for ci and other infra stuff and there it probably still works

Reply:
Seeing this, I'm not sure about the ideal approach for this. Shared grammar for both specs? and running specialized checks later at the parser & linter levels?

Reply:
Probably yes since otherwise we take away most of the reason that people would use this.